Source/Core/DolphinQt2/GameList/GameFile.cpp

```diff
@@ -417,3 +429,256 @@ QString FormatSize(qint64 size)
+{
+  return m_game_id[path];
+}
+
+QString GameFile::Cache::GetMakerID(const QString& path)
+{
+  return m_maker_id[path];
+}
+
+QString GameFile::Cache::GetMaker(const QString& path)
+{
+  return m_maker[path];
+}
+
+QMap<DiscIO::Language, QString> GameFile::Cache::GetLongMakers(const QString& path)
+{
+  return ConvertStorageLanguageMap<qint32, DiscIO::Language>(m_maker_long[path]);
+}
+
+QMap<DiscIO::Language, QString> GameFile::Cache::GetShortMakers(const QString& path)
+{
+  return ConvertStorageLanguageMap<qint32, DiscIO::Language>(m_maker_short[path]);
+}
+
+QString GameFile::Cache::GetInternalName(const QString& path)
+{
+  return m_name_internal[path];
+}
+
+QMap<DiscIO::Language, QString> GameFile::Cache::GetLongNames(const QString& path)
+{
+  return ConvertStorageLanguageMap<qint32, DiscIO::Language>(m_name_long[path]);
+}
+
+QMap<DiscIO::Language, QString> GameFile::Cache::GetShortNames(const QString& path)
+{
+  return ConvertStorageLanguageMap<qint32, DiscIO::Language>(m_name_short[path]);
+}
+
+DiscIO::Platform GameFile::Cache::GetPlatform(const QString& path)
+{
+  return static_cast<DiscIO::Platform>(m_platform[path]);
+}
+
+DiscIO::Region GameFile::Cache::GetRegion(const QString& path)
+{
+  return static_cast<DiscIO::Region>(m_region[path]);
+}
+
+DiscIO::Country GameFile::Cache::GetCountry(const QString& path)
+{
+  return static_cast<DiscIO::Country>(m_country[path]);
+}
+
+DiscIO::BlobType GameFile::Cache::GetBlobType(const QString& path)
+{
+  return static_cast<DiscIO::BlobType>(m_blob_type[path]);
+}
+
+u64 GameFile::Cache::GetRawSize(const QString& path)
+{
+  return m_raw_size[path];
+}
+
+QMap<DiscIO::Language, QString> GameFile::Cache::GetDescriptions(const QString& path)
+{
+  return ConvertStorageLanguageMap<qint32, DiscIO::Language>(m_description[path]);
+}
+
+u16 GameFile::Cache::GetRevision(const QString& path)
+{
+  return m_revision[path];
+}
+
+u8 GameFile::Cache::GetDiscNumber(const QString& path)
+{
+  return m_disc_number[path];
+}
+
+int GameFile::Cache::GetRating(const QString& path)
+{
+  return m_rating[path];
+}
+
+QString GameFile::Cache::GetApploaderDate(const QString& path)
+{
+  return m_apploader_date[path];
+}
+
+QPixmap GameFile::Cache::GetBanner(const QString& path)
+{
+  return m_banner[path];
+}
+
+void GameFile::Cache::Load()
+{
+  file.reset();
+  QDataStream stream(&file);
+  stream.setVersion(DATASTREAM_VERSION);
+
+  int ver;
+  stream >> ver;
+  stream >> m_timestamp;
+  stream >> m_title_id;
+  stream >> m_game_id;
+  stream >> m_maker_id;
+  stream >> m_maker;
+  stream >> m_maker_long;
+  stream >> m_maker_short;
+  stream >> m_name_internal;
+  stream >> m_name_long;
+  stream >> m_name_short;
+  stream >> m_platform;
+  stream >> m_region;
+  stream >> m_country;
+  stream >> m_blob_type;
+  stream >> m_raw_size;
+  stream >> m_description;
+  stream >> m_revision;
+  stream >> m_disc_number;
+  stream >> m_rating;
+  stream >> m_apploader_date;
+  stream >> m_banner;
```

Comment:
Should this not check the stream status at the end here to see if any reads failed or any other failures occurred?